action.yaml

```diff
@@ -162,15 +162,25 @@ runs:
         # This limitation affects how we handle default values and empty inputs.
         # For more information, see: https://github.com/actions/runner/issues/924
 
-        # Function to set environment variable only if the input is provided and different from default
+        # Function to set environment variable only if the input is provided, or a default required
         set_env_var_if_provided() {
           local var_name="$1"
           local input_value="$2"
           local default_value="$3"
+          local external_value="${!var_name}"
         
-          if [ ! -z "$input_value" ]; then
+          if [ -n "$external_value" ]; then
```

Comment:
I saw your changes - and looks like we need to change order.

IIUC you want to check if runner uses env - so we don't need to save this env into trivy_envs.txt, right?

But Trivy uses the followign order: flag -> env -> config file.
I think that for trivy-action - input == flag.
So we need to take value from input first.

But after your changes we take runner env first.

correct me, if i missed something.

Reply:
Yes I realised that the order I've implemented doesn't match what the Action documents, nor what was intended, currently fixing and testing that further

Reply:
Also broken my own new BATS tests so need to debug and fix those!